tiledb/sm/query/query.cc

```diff
@@ -1003,6 +1026,14 @@ Status Query::set_data_buffer(
       throw QueryStatusException("[set_data_buffer] Unsupported query type.");
     }
 
+    const bool exists = label_buffers_.find(name) != label_buffers_.end();
+    if (status_ != QueryStatus::UNINITIALIZED && !exists &&
+        !allow_separate_attribute_writes() && !serialization_allow_new_attr) {
```

Comment:
!allow_separate_attribute_writes() should be removed for dimension label checks...